src/widgets/helper/SearchPopup.cpp

```diff
@@ -126,24 +132,90 @@ void SearchPopup::updateWindowTitle()
     this->setWindowTitle("Searching in " + historyName + " history");
 }
 
+void SearchPopup::showEvent(QShowEvent *)
+{
+    this->search();
+}
+
 void SearchPopup::search()
 {
+    if (snapshot_.size() == 0)
+    {
+        this->snapshot_ = this->buildSnapshot();
+    }
+
     this->channelView_->setChannel(filter(this->searchInput_->text(),
-                                          this->channelName_, this->snapshot_,
-                                          this->channelFilters_));
+                                          this->channelName_, this->snapshot_));
+}
+
+LimitedQueueSnapshot<MessagePtr> SearchPopup::buildSnapshot()
+{
+    // no point in filtering/sorting if it's a single channel search
```

Comment:
Does this mean opening a search window in a channel that had filters applied will search outside of those filters after this PR?

Reply:
Tested this myself and it works as expected, very nice